### PR TITLE
[ROCm] Enable Tiled layout extension and minor changes to setup

### PR DESCRIPTION
**Pull request by lcskrishna**

### CODE REVIEW

**Review by petrex — COMMENTED** [review]
@lcskrishna I am seeing env issue with latest pytorch docker. Did I miss anything?

To be more specific, the following bf16 types are not converted.
        ("__nv_bfloat16", ("__hip_bfloat16", CONV_TYPE, API_RUNTIME)),
        ("__nv_bfloat162", ("__hip_bfloat162", CONV_TYPE, API_RUNTIME)),

File: torchao/csrc/cuda/tensor_core_tiled_layout/tensor_core_tiled_layout.cu [hunk 1]
```diff
@@ -123,11 +164,18 @@ __global__ void _dequantize_int4_kernel(
       // All b values within a 16x16 tile should fall within the same q group
       // Hence we load 1 scale and zero per loop
       int qgroup = ks[0] /  groupSize;
-      const __nv_bfloat16 *pSZ = reinterpret_cast<const __nv_bfloat16*>(&scales_and_zeros.value()[qgroup][n0][0]);
-
-      // Vectorize scales and zeros
-      __nv_bfloat162 scale2 = __bfloat162bfloat162(pSZ[0]);
-      __nv_bfloat162 zero2 = __bfloat162bfloat162(pSZ[1]);
+      if (scales_and_zeros) {
+        const auto&sz = *scales_and_zeros;
```
Comment (petrex) [inline torchao/csrc/cuda/tensor_core_tiled_layout/tensor_core_tiled_layout.cu:168]: @lcskrishna This change fixes compilation for me, please refer to my scratch space : https://github.com/petrex/ao/tree/rocm_tensor_tile

**Review by petrex — COMMENTED** [review]
I am also thinking about do we need micro-arch check due to subtle different in the ISAs? maybe later

**Comment by lcskrishna** [comment]
> @lcskrishna I am seeing env issue with latest pytorch docker. Did I miss anything?
> 
> To be more specific, the following bf16 types are not converted. ("__nv_bfloat16", ("__hip_bfloat16", CONV_TYPE, API_RUNTIME)), ("__nv_bfloat162", ("__hip_bfloat162", CONV_TYPE, API_RUNTIME)),

Ah!! forgot to mention, use pytorch nightly. This seems to work after PT 2.5.

File: torchao/csrc/cuda/tensor_core_tiled_layout/tensor_core_tiled_layout.cu (hunk 1 above)
Comment (lcskrishna) [reply torchao/csrc/cuda/tensor_core_tiled_layout/tensor_core_tiled_layout.cu:168]: Thanks @petrex for taking a look. I will update it accordingly.

**Comment by lcskrishna** [comment]
> I am also thinking about host side micro-arch check due to subtle differences in the ISAs? maybe later

Yes let's seperate that out and do it later.

**Comment by petrex** [comment]
could you share the docker image you are using? and maybe a little clean up on the code. thx

**Pull request merged**